public/javascripts/SVValidate/src/data/Form.js

```diff
@@ -3,23 +3,38 @@ function Form(url) {
         dataStoreUrl : url
     };
 
+    /**
+     * Compiles data into a format that can be parsed by our backend.
+     * @returns {{}}
+     */
     function compileSubmissionData() {
         var data = {};
-        var mission = svv.missionContainer.getCurrentMission();
+        var missionContainer = svv.missionContainer;
+        var mission = missionContainer ? missionContainer.getCurrentMission() : null;
 
-        data.interactions = svv.tracker.getActions();
-        data.labels = svv.labelContainer.getCurrentLabels();
+        var labelContainer = svv.labelContainer;
+        var labelList = labelContainer ? labelContainer.getCurrentLabels() : null;
 
-        // Add the current mission
-        data.missionProgress = {
-            mission_id: mission.getProperty("missionId"),
-            labels_progress: mission.getProperty("labelsProgress"),
-            completed: mission.getProperty("completed"),
-            skipped: mission.getProperty("skipped")
-        };
+        // Only submit mission progress if there is a mission when we're compiling submission data.
+        if (mission) {
+            // Add the current mission
+            data.missionProgress = {
+                mission_id: mission.getProperty("missionId"),
+                labels_progress: mission.getProperty("labelsProgress"),
+                label_type_id: mission.getProperty("labelTypeId"),
+                completed: mission.getProperty("completed"),
+                skipped: mission.getProperty("skipped")
+            };
+        }
 
+        // Only label list if there is a label list when we're compiling submission data.
+        if (labelList) {
+            data.labels = svv.labelContainer.getCurrentLabels();
```

Comment:
This is the part I was talking about in the very beginning. Probably just want data.labels to be an empty list instead of null if there are no labels.